enterprise/dev/deployment-notifications/deployments_notifier.go

```diff
@@ -0,0 +1,174 @@
+	ErrAlreadyDeployed = errors.New("target commit matches live commit")
+)
+
+type DeploymentNotifier struct {
+	vr           VersionRequester
+	ghc          *github.Client
+	changedFiles []string
+	targetCommit string
+}
+
+func NewDeploymentNotifier(ghc *github.Client, vr VersionRequester, targetCommit string, changedFiles []string) *DeploymentNotifier {
+	if targetCommit == "" {
+		panic("can't operate with a blank target commit")
+	}
+	return &DeploymentNotifier{
+		vr:           vr,
+		ghc:          ghc,
+		changedFiles: changedFiles,
+		targetCommit: targetCommit,
+	}
+}
+
+// getNewCommits returns a slice of commits starting from the target commit up to the currently deployed commit.
+func (dn *DeploymentNotifier) getNewCommits(ctx context.Context, lastCommit string) ([]*github.RepositoryCommit, error) {
+	var page = 1
+	var commits []*github.RepositoryCommit
+	for page != 0 && page != maxCommitsPageCount {
+		cs, resp, err := dn.ghc.Repositories.ListCommits(ctx, repoOwner, repoName, &github.CommitsListOptions{
+			SHA: "main",
+			ListOptions: github.ListOptions{
+				Page:    page,
+				PerPage: commitsPerPage,
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+		commits = append(commits, cs...)
+		var currentCommitIdx int
+		for i, commit := range commits {
+			if strings.HasPrefix(commit.GetSHA(), dn.targetCommit) {
+				currentCommitIdx = i
+			}
+			if commit.GetSHA() == lastCommit {
+				return commits[currentCommitIdx:i], nil
+			}
+		}
+		page = resp.NextPage
+	}
+	return nil, errors.Newf("commit %s not found in the last %d commits", lastCommit, maxCommitsPageCount*commitsPerPage)
+}
+
+func (dn *DeploymentNotifier) getNewPullRequests(ctx context.Context, liveCommit string) ([]*github.PullRequest, error) {
+	repoCommits, err := dn.getNewCommits(ctx, liveCommit)
+	if err != nil {
+		return nil, err
+	}
+	var pullsSinceLastCommit []*github.PullRequest
+	for _, rc := range repoCommits {
+		pulls, _, err := dn.ghc.PullRequests.ListPullRequestsWithCommit(
+			ctx,
+			repoOwner,
+			repoName,
+			rc.GetSHA(),
+			&github.PullRequestListOptions{
+				State: "merged",
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+		pullsSinceLastCommit = append(pullsSinceLastCommit, pulls...)
+	}
+	return pullsSinceLastCommit, nil
+}
+
+func (dn *DeploymentNotifier) Report(ctx context.Context) (*report, error) {
+	liveCommit, err := dn.vr.LastCommit()
+	if err != nil {
+		return nil, err
+	}
+
+	if liveCommit == dn.targetCommit {
+		return nil, errors.Wrap(ErrAlreadyDeployed, dn.vr.Environment())
+	}
+
+	prs, err := dn.getNewPullRequests(ctx, liveCommit)
+	if err != nil {
+		return nil, err
+	}
+
+	report := report{
+		Environment:       dn.vr.Environment(),
+		PullRequests:      prs,
+		DeployedAt:        time.Now().In(time.UTC).Format(time.RFC822Z),
+		Apps:              dn.deployedApps(),
+		BuildkiteBuildURL: os.Getenv("BUILDKITE_BUILD_URL"),
+	}
+
+	return &report, nil
+}
+
+func (dn *DeploymentNotifier) deployedApps() []string {
```

Comment:
```suggestion
func (dn *DeploymentNotifier) deployedServices() []string {
```

We refer to these as `services` everywhere else - I've never seen anyone refer to frontend or gitserver as an "app"